lightstep/resource_metric_condition.go

```diff
@@ -650,6 +652,15 @@ func buildQueries(queriesIn []interface{}) ([]client.MetricQueryWithAttributes,
 		}
 		newQueries = append(newQueries, newQuery)
 	}
+
+	if hasSpanSingle {
+		for i := 0; i < len(newQueries); i++ {
+			if newQueries[i].Type == "composite" {
+				newQueries[i].Type = "spans_composite"
+			}
```

Comment:
will things get weird if there's a mix of metrics and spans queries?

Reply:
AFAIK, you can't actually mix span and metric queries. I just double checked this is the case in the UI. I think if we wanted to correctly fix this in the future we would need to not have a different composite query type for spans and metrics. They have no user facing difference anyway.

Reply:
I agree someone shouldn't be able to and the UI prevents it but can someone write a terraform file to mix them? Then my concern is here we override one of the types in a way that might result in us sending in weird data that makes it through the public API when it shouldn't

Reply:
I just tried it and it fails validation on the server side if you try and mix metric and span queries in a formula.

Reply:
thank you for checking that!